File: in_Java/1200 Minimum Absolute Difference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        List<List<Integer>> res = new ArrayList<>();
        Arrays.sort(arr);
        int minlen = Integer.MAX_VALUE;
        for(int i = 1; i < arr.length; i++){
            if(arr[i] - arr[i-1] < minlen) minlen = arr[i] - arr[i-1];
        }
        for(int i = 1; i < arr.length; i++){
            List<Integer> temp = new ArrayList<>();
            if(arr[i] - arr[i-1] == minlen){
                temp.add(arr[i-1]);
                temp.add(arr[i]);
                res.add(temp);
            }
        }
        return res;
    }
}

File: in_Python/1200 Minimum Absolute Difference.py
class Solution(object):
    def minimumAbsDifference(self, arr):
        """
        :type arr: List[int]
        :rtype: List[List[int]]
        """
        arr1 = sorted(arr)
        mindis = 10000000
        for i in range(1,len(arr)):
            curd = arr1[i] - arr1[i-1]
            if abs(curd) < mindis:
                mindis = abs(curd)
        res = []
        arrs = set(arr)
        for i in range(len(arr)):
            x = arr1[i]
            if x + mindis in arrs:
                y = x + mindis
                temp = []
                temp.append(x)
                temp.append(y)
                res.append(temp)
        return res
        